docs/getting_started/execution-modes.rst

```diff
@@ -72,7 +72,10 @@ In this case, users are responsible for declaring which version of ``dbt`` they
 
 Similar to the ``local`` execution mode, Cosmos converts Airflow Connections into a way ``dbt`` understands them by creating a ``dbt`` profile file (``profiles.yml``).
 
-A drawback with this approach is that it is slower than ``local`` because it creates a new Python virtual environment for each Cosmos dbt task run.
+Some drawbacks of this approach:
+
+- It is slower than ``local`` because it creates a new Python virtual environment for each Cosmos dbt task run.
```

Comment:
This will change once we merge #611 🙌

Reply:
It depends on whether the venv is saved somewhere though, right? So this language change will still apply in some cases, i.e. if the flag is not set.

(Perhaps a separate discussion, but at the moment, my current setup is that I already run dbt out of a separate virtualenv, I just point the binary to `dbt_venv/bin/dbt`. I'm curious what the pros/cons are to saving the venv compared to that approach.)

Reply:
> It depends on whether the venv is saved somewhere, though, right? So, this language change will still apply in some cases, i.e. if the flag is not set.

Yes, you're right! It will also depend on how Airflow was configured. If the Airflow worker nodes run using `KubernetesExecutor`, the optimisation would only work if the dbt binary was in a shared volume - which could lead to concurrency issues.

> (Perhaps a separate discussion, but at the moment, my current setup is that I already run dbt out of a separate virtualenv, I point the binary to dbt_venv/bin/dbt. I'm curious what the pros/cons are to saving the venv and using the virtualenvoperator, compared to that approach of using local but pointing to a separate venv's bin/dbt.)

I'd encourage users to use `LoadMode.LOCAL`. It's the most tested approach, and - since the latest dbt and Airflow dependencies don't conflict (https://astronomer.github.io/astronomer-cosmos/getting_started/execution-modes-local-conflicts.html#execution-modes-local-conflicts) - this is the most straightforward setup.

If there are dependency conflicts, the second recommendation would be to use `ExecutionMode.LOCAL` customising the dbt binary path, as you mentioned.

I'm curious about your use case. Are you using Google Cloud Composer? How are you setting up the dbt binary? Are you adding to the shared volume/GCS bucket?

The `ExecutionMode.VIRTUALENV` is an alternative for users who may not be able to create a separate venv for dbt in the Airflow worker nodes and who are facing dependency conflicts but do not want the overhead of using `ExecutionMode.KUBERNETES`.

Overall, each ExecutionMode has its trade-offs.

Reply:
>I'm curious about your use case. Are you using Google Cloud Composer? How are you setting up the dbt binary? Are you adding to the shared volume/GCS bucket?

I am not using GCC at the moment. I am using Astronomer and I pip install into a new venv as part of the Docker build step. Locally I have a command that aliases `dbt` to `rich-click venv_dbt/bin/dbt` and passes in a default project dir / profile dir and default port for docs.